Week-5/exercise_5_2/index.js

```diff
@@ -0,0 +1,37 @@
+// function isVowel(char) {
+//   return 'aeiou'.includes(char);
+// }
+
+// function vowelCount(str) {
+//   const vowelMap = new Map();
+//   for (let char of str) {
+//     let lowerCaseChar = char.toLowerCase();
+//     if (isVowel(lowerCaseChar)) {
+//       if (vowelMap.has(lowerCaseChar)) {
+//         vowelMap.set(lowerCaseChar, vowelMap.get(lowerCaseChar) + 1);
+//       } else {
+//         vowelMap.set(lowerCaseChar, 1);
+//       }
+//     }
+//   }
+//   return vowelMap;
+// }
+
+function vowelCount(str) {
+  const vowels = ['a', 'e', 'i', 'o', 'u'];
+  const counts = new Map();
+
+  for (let i = 0; i < str.length; i++) {
+    const char = str[i].toLowerCase();
+    if (vowels.includes(char)) {
```

Comment:
this is O(n). Instead we could make a map of vowels and keep track of count there which will make it O1

Reply:
I already had used Map to keep track of vowels count